### PR TITLE
working hoverbox in chart gaps

### PR DESCRIPTION
**Pull request by aaronz8**
Fixed visualizations to show hoverbox for null values in graph.
I also changed the line graph example to show a null value.
To test: http://localhost:3000/examples/visualizations/line.html

### CODE REVIEW

**Comment by aaronz8** [comment]
screen shot 2015-03-12 at 4 32 57 pm

@drakonen @Juice10

**Comment by aaronz8** [comment]
Actually, just the line and motion scatter charts work with null values. :( Will work on fixing the other chart types today/tmrw

File: src/visualization/hover-mixin.js [hunk 1]
```diff
@@ -171,7 +171,7 @@ define(['../core', './lib/custom-hover-details', '../lib/utils'], function(Openh
             } else {
               label += '<tr class="hover-item">';
               label += '  <td><span class="hover-label" style="color: ' + line.color + '"> ' + (line.lineLabel || line.name) + ' </span></td>';
-              label += '  <td>' + yFormat(line.data[xPos].y);
+              label += '  <td>' + (line.data[xPos].y !== null ? yFormat(line.data[xPos].y) : "-");
```
Comment (kevinvandijk) [inline src/visualization/hover-mixin.js]: Please use either single or double quotes, but not mixed (the `' <td>'` and then  `"-"`)

File: src/visualization/lib/custom-hover-details.js [hunk 2]
```diff
@@ -183,7 +183,8 @@ define(['rickshaw', 'core'], function (Rickshaw, Openhose) {
         }
       }
 
-      if (point.value.y === null) return;
+      // Uncomment next line to hide hoverbox for null values
+      //if (point.value.y === null) return;
```
Comment (kevinvandijk) [inline src/visualization/lib/custom-hover-details.js]: Why not just remove it instead of commenting it? Will it be needed for some specific things?

**Comment by aaronz8** [comment]
cleaned up code: removed unnecessary comment and unmixed ' and ".

**Comment by drakonen** [comment]
on the motion scatter I get a lot of errors in the console:
`Error: Invalid negative value for <circle> attribute r="-46.98122325410294"`

**Comment by aaronz8** [comment]
Fixed. There was some weird behavior with motion scatter when there's null data, such that the dot would jump and then animate to disappear. I just made it do the hide animation at the previous position instead of jumping.

File: dist/openhose-visualizations-sdk.core.js [hunk 3]
```diff
@@ -752,8 +752,6 @@ visualization_lib_custom_hover_details = function (Rickshaw, Openhose) {
           }
         }
       }
-      if (point.value.y === null)
```
Comment (drakonen) [inline dist/openhose-visualizations-sdk.core.js]: Next time, do not commit the bundles.

**Comment by drakonen** [comment]
:+1:

File: dist/openhose-visualizations-sdk.core.js (hunk 3 above)
Comment (aaronz8) [inline dist/openhose-visualizations-sdk.core.js]: Oh, should I add it to the .gitignore?

Comment (drakonen) [inline dist/openhose-visualizations-sdk.core.js]: No. we need it for master when making a release etc.

**Pull request merged**